lib/cashu/secret.ts

```diff
@@ -0,0 +1,59 @@
+import {
+  type NUT10SecretData,
+  type ParsedNUT10Secret,
+  type ProofSecret,
+  WELL_KNOWN_SECRET_KINDS,
+} from './types';
+
+/**
+ * Parse secret string from Proof.secret into a well-known secret [NUT-10](https://github.com/cashubtc/nuts/blob/main/10.md)
+ * or a string [NUT-00](https://github.com/cashubtc/nuts/blob/main/00.md)
+ */
+export const parseSecret = (secret: string): ProofSecret => {
+  let parsed: unknown;
+  try {
+    parsed = JSON.parse(secret);
+  } catch {
+    // If JSON parsing fails, assume it's a plain string secret
+    // as defined in NUT-00
+    return secret;
+  }
+
+  //  Check if parsed value matches well-known secret format:
+  //      ["P2PK", {
+  //        nonce: string,
+  //        data: string,
+  //        tags?: string[]
+  //      }]
+  const isValidSecret = (value: unknown): value is ParsedNUT10Secret =>
```

Comment:
I wonder if we should start using [zod](https://zod.dev/)

it's cumbersome to write these validations manually

Reply:
I added zod for validating NUT-10 secrets.

See [this](https://github.com/MakePrisms/boardwalkcash/pull/215#discussion_r1894788024) comment.
Another option would be to get rid of the `NUT10SecretData` and `ParsedNut10Secret` and then just define these in the zod schemas. But it feels better to me to keep the types and then use `satisfies z.ZodType<NUT10SecretData>;` to make sure our schema is correct.

Reply:
I think we should define them with zod schemas and defer the types from that. I am not sure I see the benefits of the current approach and it makes you maintain the same thing in two places

Reply:
see my approach in #233 

I just removed those types for now